test/whitelisted_crowdsale.js

```diff
@@ -73,3 +76,77 @@ contract('WhitelistedCrowdsale', async (accounts) => {
   });
 
 });
+
+contract('WhitelistedCrowdsale', async (accounts) => {
+  let owner = accounts[0];
+  let holder = accounts[1];
+  let escrow_account = accounts[2];
+  let user = accounts[3];
+  let userBalance = web3.eth.getBalance(user);
+  let defaultRateForPrivateOffer = 14000;
+
+  let controller = null;
+  let instanceWhitelistedCrowdsale = null;
+  let token = null;
+
+  beforeEach("setup contract for each test", async () => {
+    let startTime = Math.ceil(Date.now() / 1000);
+    let endTime = Math.ceil(Date.now() / 1000) + 100;
+    
+    controller = await ICO_controller.new(holder, escrow_account, {from: owner});
+    token = MFC_Token.at(await controller.token.call());
+    await controller.startPrivateOffer.sendTransaction(startTime, endTime, escrow_account, {from: owner});
+    instanceWhitelistedCrowdsale = WhitelistedCrowdsale.at(await controller.privateOffer.call());
+    wait(5);
```

Comment:
Do we need wait here? Can we use smaller delay?

Reply:
We need wait here, because we have in ICO_controller.start_ICO a restriction (require(_startTime >= now); ) and on the other hand we have another restriction in Whitelisted_crowdsale.validPurshase (bool withinPeriod = now >= startTime && now <= endTime).

Reply:
@avgoeid can we use `wait(2)` instead of `wait(5)`?